components/configuration-accessor/default/jest-argv.mjs

```diff
@@ -0,0 +1,114 @@
+  JSON: { stringify: stringifyJSON, parse: parseJSON },
+  Object: { entries: toEntries, fromEntries },
+  Array: { isArray },
+} = globalThis;
+
+const { default: minimist } = Minimist;
+
+const hook = convertFileUrlToPath(
+  toAbsoluteUrl("lib/node/transformer-jest.mjs", self_directory),
+);
+
+const extractTransformEntryValue = (value) => {
+  if (typeof value === "string") {
+    return { specifier: value, options: {} };
+  } else {
+    assert(
+      !logErrorWhen(
+        !isArray(value) || value.length !== 2 || typeof value[0] !== "string",
+        "Invalid transform field, expected transform field to be either a string or an array of length two whose first element is a string: %j",
+        value,
+      ),
+      "Invalid transform field",
+      ExternalAppmapError,
+    );
+    return { specifier: value[0], options: value[1] };
+  }
+};
+
+const compileHookTransformEntry = (root) => {
+  const replacement = constant(convertFileUrlToPath(root));
+  return ([key, value]) => {
+    const { specifier, options } = extractTransformEntryValue(value);
+    return [
+      key,
+      {
+        specifier: specifier.replace(/<rootDir>\//gu, replacement),
+        options,
+      },
+    ];
+  };
+};
+
+const hookTransformObject = (transform, root) => ({
+  "^": [
+    hook,
+    fromEntries(toEntries(transform).map(compileHookTransformEntry(root))),
+  ],
+});
+
+export const hookJestArgvAsync = async (argv, base) => {
+  const options = minimist(argv);
+  const root = hasOwnProperty(options, "rootDir")
+    ? toDirectoryUrl(toAbsoluteUrl(options.rootDir, base))
+    : base;
+  if (hasOwnProperty(options, "transform")) {
+    const index = argv.indexOf("--transform");
+    assert(
+      !logErrorWhen(
+        index !== argv.lastIndexOf("--transform"),
+        "Jest `--transform` argument should not be duplicate: %j",
+        argv,
+      ),
+      ExternalAppmapError,
+      "Jest --transform argument should not be duplicate",
+    );
+    assert(
+      !logErrorWhen(
+        index === argv.length - 1,
+        "Jest `--transform` argument should not be in last postion: %j",
+        argv,
+      ),
+      ExternalAppmapError,
+      "Jest `--transform` argument should not be in last postion",
+    );
+    let transform = argv[index + 1];
+    try {
+      transform = parseJSON(transform);
+    } catch (error) {
+      logError(
+        "Jest `--transform` argument should be a json string: %j >> %O",
+        transform,
+        error,
+      );
+      throw new ExternalAppmapError(
+        "Jest --transform argument should be a json string",
+      );
+    }
+    return [
+      ...argv.slice(0, index + 1),
+      stringifyJSON(hookTransformObject(transform, root)),
+      ...argv.slice(index + 2, argv.length),
+    ];
+  } else {
+    const config = await loadJestConfigAsync(options, { root, base });
+    const transform = hasOwnProperty(config, "transform")
+      ? config.transform
+      : { "\\.[jt]sx?$": "babel-jest" };
```

Comment:
Where does this come from? Is it jest default? Is there a way to not hardcode it? This could use a comment at least.